src/main/java/com/eppo/sdk/deserializer/BanditsDeserializer.java

```diff
@@ -0,0 +1,131 @@
+package com.eppo.sdk.deserializer;
+
+import com.eppo.sdk.dto.*;
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.databind.DeserializationContext;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.deser.std.StdDeserializer;
+
+import java.io.IOException;
+import java.time.Instant;
+import java.util.*;
+
+/**
+ * Eppo Value Deserializer Class
+ */
+public class BanditsDeserializer extends StdDeserializer<Map<String, BanditParameters>> {
```

Comment:
I rolled my own deserializer because we have many `Map`s, and also for these maps I still wanted the values to know their key.

See [`bandits-paraemters-1.json`](https://github.com/Eppo-exp/java-server-sdk-bandit-beta/pull/3/files#diff-57812398e9d2358377168e236f99ba934e2d20a6a01727673965263e4476b1be) for an example of what we're deserializing into Java objects.

GitHub presents files in alphabetical order, which is not the most helpful. The big picture structure is this:

* **BanditParametersResponse**
  * `updatedAt`: `Date`
  * `bandits`: `Map<String, BanditParameters>`
     * **BanditParameters**
       * `banditKey`: `String`
       * `updatedAt`: `Date`
       * `model`: `String`
       * `modelVersion`: `String`
       * `modelData`: `BanditModelData`
         *  **BanditModelData**
           * `gamma`: `Double`
           * `coefficients`: `Map<String, BanditCoefficients>`
             * **BanditCoefficients**
               * `actionKey`: `String`
               * `intercept`: `Double`
               * `subjectNumericCoefficients`: `Map<String, BanditNumericAttributeCoefficients>`
               * `subjectCategoricalCoefficients`: `Map<String, BanditCategoricalAttributeCoefficients>`
               * `actionNumericCoefficients`: `Map<String, BanditNumericAttributeCoefficients>`
               * `actionCategoricalCoefficients`: `Map<String, BanditCategoricalAttributeCoefficients>`
                 * **BanditNumericAttributeCoefficients**
                   * `attributeKey`: `String`
                   * `coefficient`: `Double`
                   * `missingValueCoefficient`: `Double`
                 * **BanditCategoricalAttributeCoefficients**
                    * `attributeKey`: `String`
                   * `coefficient`: `Double`
                   * `valueCoefficients`: `Map<String, Double>`

Reply:
I didn't check the Java code in detail. I'll do a line-by-line pass after we align on the points above.

Reply:
@aarsilv Logic wise looks good to me, I get the point why you want a separate Deserializer. Just one point, can't we send a payload something like? Which has duplicate value but it doesn't required a dedicated deserializer. 
```
{'key-1': {key: 'key-1', .....}}
```
instead of 
```
[{'key': 'key-1', .....}]
```

Reply:
@giorgiomartini0 
1. yes `banditKey` and `actionKey` are the same values as the keys in their containing maps.
2. happy to rename `model` to `modelName` if it is causing confusion
3. Unclear what we'll send as `modelVersion`--sending the id seems like the simplest although its unique across all models. Made a string so we could do things like semantic versioning if we wanted. We could instead change to `modelId` and drop the "version" concept.
4. This is how it is--`missingValueCoefficient` where I had `coefficient`. copy-paste typo which I fixed in the comment.

Reply:
@agamkrbit that was what I originally envisioned but Evan suggested having arrays instead of maps based on past pain points and friction he's encountered--and I trust his experience.

Reply:
1. 👍 I just noticed now that you called it out in the first sentence of the comment. Makes sense.
2. The name is a bit ambiguous but no need to rename. We already use `model` that way elsewhere.
3. I like the flexibility of a string `version`. It'll be easy to make these into ids if needed; the other way is impossible.
4. Looks good!